core/src/main/scala/org/apache/spark/sql/execution/joins/LocalJoin.scala

```diff
@@ -335,10 +410,15 @@ case class LocalJoin(leftKeys: Seq[Expression],
           streamSideKeys.map(BindReferences.bindReference(_, left.output)))
     }
     val streamKeyVars = ctx.generateExpressions(streamKeys)
-    mapAccessor.generateMapLookup(entryVar, localValueVar, keyIsUniqueTerm,
+
+    val mapAccesCode = mapAccessor.generateMapLookup(entryVar, localValueVar, keyIsUniqueTerm,
       numRowsTerm, nullMaskVars, initCode, checkCondition,
       streamSideKeys, streamKeyVars, buildKeyVars, buildVars, input,
       resultVars, dictionaryArrayTerm, joinType)
+    s"""
+       $evaluatedInputVars
+       $mapAccesCode
+     """
```

Comment:
Need to check any effect on dictionary optimization that avoids materializing the string key (unless its a miss in the current dictionary array). Hopefully JIT will optimize out unused variable. Can you check what numbers you see with TPCETrade test after increasing the sizes by 10X to 34M and 5M? Also enable the RDD delinking flag temporarily. Normally I get <200ms best case 160-170 sometimes 190. If there is any change, then let this change go in but we will need to track the issue.